### PR TITLE
new UT_Signage_1

### PR DESCRIPTION
**Pull request by javierNadal**
# IFC4.3 Unit test

##  WIP: UT_Signage_1
This is a proposal of first unit test por Signaling.
I have not found a clear example in story lines, so I think this could a stating point.

@pjanck

### CODE REVIEW

File: IFC 4.3/Align_1/UT_Align_1.IFC [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+HEADER;
+FILE_DESCRIPTION(('ViewDefinition [CoordinationView]'),'2;1');
+FILE_NAME('UT_Align_1.IFC','2020-11-16T12:56:17',(),(),'IfcOpenShell 0.5.0-dev','IfcOpenShell 0.5.0-dev','');
+FILE_SCHEMA(('IFC4X3_RC2'));
+ENDSEC;
+DATA;
+#1=IFCACTORROLE(.USERDEFINED.,'',$);
+#2=IFCADDRESS(.HOME.,'',$);
+#3=IFCPERSON('Javier','','',$,$,$,(#1),(#2));
+#4=IFCADDRESS(.OFFICE.,'',$);
```
Comment (jmirtsch) [inline IFC 4.3/Align_1/UT_Align_1.IFC:14]: An IfcAddress is an abstract class, and only classes inheriting from this should be instantiated such as IfcPostalAddress.  Can you please adjust?  I believe IfcOpenShell has a new audit feature to detect this when reviewing the ifc file if it helps.

File: IFC 4.3/Align_1/UT_Align_1.IFC [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+/* Empty road */
+#32=IFCROAD('1FXfkfy6b93BwiEai_gN$E',$,'None',$,'IfcRoad',$,$,$,.COMPLEX.);
+
+/* Road in project */
+#33=IFCRELAGGREGATES('2uInlzG_X1_glgN9RNs$c8',#9,$,$,#30,(#32));
+
+/* Alignment: horizontal and vertical alignment */
+#34=IFCALIGNMENT('0XTxY6KDrCdQRFzSOb6jwJ',$,'UT_Align_1',$,$,$,$,#35,$);
+#35=IFCALIGNMENTCURVE(#36,#73,$);
```
Comment (jmirtsch) [inline IFC 4.3/Align_1/UT_Align_1.IFC:56]: IfcAlignmentCurve is now deprecated in RC2, https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/link/ifcalignmentcurve.htm  The horizontal, vertical and cant alignment concepts are now aggregated into an IfcAlignment using IfcRelAggregates relationship.

File: IFC 4.3/Align_1/UT_Align_1.IFC [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+/* Horizontal alignment */
+#36=IFCALIGNMENTHORIZONTAL('06__Wngff6zAUzQvB4Fnt4',$,$,$,$,$,$,0.,(#60,#38,#40,#42,#44,#46,#62,#48,#50,#52,#64,#54,#56,#58,#66));
+#37=IFCCARTESIANPOINT((335095.95046541,4084593.74863239));
+#38=IFCALIGNMENTHORIZONTALSEGMENT($,$,#37,357.802095555,-25.,-25.,39.84063746,$,.CIRCULARARC.);
```
Comment (jmirtsch) [inline IFC 4.3/Align_1/UT_Align_1.IFC:61]: Negative radius of curvature indicates right hand turning segment, where as image and next segment start point indicates left hand turn.  I believe all radius values in the sample should be multiplied by negative 1.

File: IFC 4.3/Align_1/UT_Align_1.IFC [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+#64=IFCALIGNMENTHORIZONTALSEGMENT($,$,#63,75.983268735,0.,0.,12.39520568,$,.LINE.);
+#65=IFCCARTESIANPOINT((335385.54643726,4084683.81177391));
+#66=IFCALIGNMENTHORIZONTALSEGMENT($,$,#65,9.832185999,0.,0.,35.3941232,$,.LINE.);
+
+/* Alignment in Site */
+#67=IFCRELCONTAINEDINSPATIALSTRUCTURE('3DvGg4YE9DwRm6tfQg$num',$,$,$,(#34),#30);
+
+/* Vertical alignment */
+#68=IFCALIGNMENTVERTICALSEGMENT($,$,0.,14.256,365.8,0.0784810126582277,0.0784810126582277,$,.CONSTANTGRADIENT.);
+#69=IFCALIGNMENTVERTICALSEGMENT($,$,14.256,129.488,366.918825316456,0.0784810126582277,0.0784810126582277,890.,.PARABOLICARC.);
```
Comment (jmirtsch) [inline IFC 4.3/Align_1/UT_Align_1.IFC:96]: A vertical segment that isn't constant should provide distinct start and end gradients.  Please review and adjust.

File: IFC 4.3/Align_1/UT_Align_1.IFC (hunk 1 above)
Comment (javierNadal) [reply IFC 4.3/Align_1/UT_Align_1.IFC:14]: Hello
First, thank you very much for your comment about audit feature of IfcOpenShell. I will look for it.

I am changing this isssue, and others you told me, but I would like also to include last added attribute to IfcCurveSegment, as @SergejMuhic presented yesterday. 
I see that https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/link/ifccurvesegment.htm is not updated.
Please, do you know where can I see last changes? In miro board? github? 

Thank you

File: IFC 4.3/Align_1/UT_Align_1.IFC (hunk 4 above)
Comment (javierNadal) [reply IFC 4.3/Align_1/UT_Align_1.IFC]: Fixed in 1954276

File: IFC 4.3/Align_1/UT_Align_1.IFC (hunk 3 above)
Comment (javierNadal) [reply IFC 4.3/Align_1/UT_Align_1.IFC]: Fixed in 1954276

File: IFC 4.3/Align_1/UT_Align_1.IFC (hunk 2 above)
Comment (javierNadal) [reply IFC 4.3/Align_1/UT_Align_1.IFC]: Fixed in 1954276

File: IFC 4.3/Align_1/UT_Align_1.IFC (hunk 1 above)
Comment (javierNadal) [reply IFC 4.3/Align_1/UT_Align_1.IFC]: Fixed in 1954276

**Comment by pjanck** [comment]
We're moving to the new repository: https://github.com/bSI-InfraRoom/IFC-infra-unit-test

Please open a new pull request there and close this one.

**Pull request closed without merging**